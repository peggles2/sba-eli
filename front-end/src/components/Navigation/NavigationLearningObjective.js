import React, { Component } from "react";
import { Link } from 'react-router-dom';
import { Divider, List } from "semantic-ui-react";
import axios from "axios";

export default class NavigationLearningObjective extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningObjectives: []
    };
  }

  handleItemClick = (e, {name}) => this.setState({
    activeItem: name
  });

  componentDidMount() {
    this.fetchData();
  }

  fetchData() {
    const url =
      process.env.REACT_APP_SERVICE_HOST +
      "/learning_objectives/" +
      "?course_id=" +
      this.props.learningPathId;

    axios
      .get(url)
      .then(res => {
        const learningObjectives = res.data;
        this.setState({ learningObjectives });
      })
      .catch(error => {
        console.error(error);
      });
  }

  topicNumber() {
    const objectives = this.state.learningObjectives || [];
    const topicNumber = objectives.length;

    if (topicNumber === 1) {
      return topicNumber + " Topic";
    } else {
      return topicNumber + " Topics";
    }
  }

  render() {
    const learningObjectivePath = `/learning_paths/${this.props.learningPathId}/learning_objectives/`
    const objectives = this.state.learningObjectives || [];
    const topics = objectives.map((lo, index) => (
      <List.Item key={'learning_objective_' + index}><Link to={learningObjectivePath + lo.id} onClick={this.handleItemClick}>{lo.name}</Link></List.Item>
    ));

    return (
      <div className='topic-summary'>
        <em>{this.topicNumber()}, 1 hour 24 minutes</em><br/>
        <p className='path-description'>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.</p>
        <Divider />
        { topics }
      </div>
    )
  }
}
