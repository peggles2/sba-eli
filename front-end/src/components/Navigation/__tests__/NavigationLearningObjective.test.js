import React from 'react';
import NavigationLearningObjective from '../NavigationLearningObjective'
import { shallow, mount } from 'enzyme';

describe('NavigationLearningObjective', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<NavigationLearningObjective/>);
    
    expect(wrapper).toMatchSnapshot();
  });

});
