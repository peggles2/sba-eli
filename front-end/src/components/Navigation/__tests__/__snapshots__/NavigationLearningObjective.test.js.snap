// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationLearningObjective should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationLearningObjective />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <em>
          0 Topics
          , 1 hour 24 minutes
        </em>,
        <br />,
        <p
          className="topic-description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.
        </p>,
        <Divider />,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "0 Topics",
            ", 1 hour 24 minutes",
          ],
        },
        "ref": null,
        "rendered": Array [
          "0 Topics",
          ", 1 hour 24 minutes",
        ],
        "type": "em",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.",
          "className": "topic-description",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <em>
            0 Topics
            , 1 hour 24 minutes
          </em>,
          <br />,
          <p
            className="topic-description"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.
          </p>,
          <Divider />,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "0 Topics",
              ", 1 hour 24 minutes",
            ],
          },
          "ref": null,
          "rendered": Array [
            "0 Topics",
            ", 1 hour 24 minutes",
          ],
          "type": "em",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.",
            "className": "topic-description",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec eu viverra dolor. In pharetra non nunc vitae cursus. Donec fermentum vestibulum orci ut aliquam. Phasellus eu arcu scelerisque, pretium massa eget, semper justo.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
